src/engine/renderer/tr_image.cpp

```diff
@@ -1361,6 +1361,24 @@ void R_UploadImage( const byte **dataArray, int numLayers, int numMips,
 		ri.Hunk_FreeTempMemory( scaledBuffer );
 	}
 
+	// detect heightmap in normalmap alpha channel
+	// and enable it
+	if ( image->bits & IF_NORMALMAP )
```

Comment:
I just thought that we may first test if we already know if there is an heightmap or not (thanks to explicit shader keyword) before trying to detect its presence:

```c
if ( image->bits & IF_NORMALMAP || !( image->bits & IF_DISPLACEMAP) )
```c